pandas/_libs/properties.pyi

```diff
@@ -0,0 +1,3 @@
+
+cache_readonly = property
```

Comment:
I think this will need many ignore statements, since there are quite a few places where a typed `cache_readonly` will reveal type issues (the type issues are already there, but mypy doesn't know about them because it couldn't even find `cache_readonly` until now).

I think you will also need to add annotations for `AxisProperty` as that is important elsewhere. It would be great to have partial type-stubs.